runtime/shared_common/CacheMap.cpp

```diff
@@ -216,14 +216,17 @@ SH_CacheMap::newInstance(J9JavaVM* vm, J9SharedClassConfig* sharedClassConfig, S
 void
 SH_CacheMap::dontNeedMetadata(J9VMThread* currentThread) 
 {
-	/* Local copies to avoid race condition */
 	Trc_SHR_CM_j9shr_dontNeedMetadata(currentThread);
+	SH_CompositeCacheImpl* ccToUse = _ccHead;
 
+	if (_metadataReleased) {
```

Comment:
Is there any guarantee this function will only be called once? I'm a little concerned about changing the behavior and releasing less memory after these changes than were released before these changes. If this is called multiple times, the original code tracked updated bounds in some cases and released the larger area.

Reply:
This function can be called more than once by multiple threads that detect vm phase change. The behaviors of original code is:
1. warm run: updateAccessedShrCacheMetadataBounds() is called only when _metadataReleased is false. So even though dontNeedMetadata() can be call more than once, max/minAccessedShrCacheMetadata won' be updated after the first dontNeedMetadata() call. So we were releasing the same region multiple times.

2. cold run: the cold run code was added by https://github.com/eclipse/openj9/pull/3454. The check of _metadataReleased was not added before calling updateAccessedShrCacheMetadataBounds() there. The behavior is inconsistent with warm run. 

This commit adds check of _metadataReleased before calling updateAccessedShrCacheMetadataBounds() in the cold run, so that it is consistent with warm run. It might be possible in a cold run that a smaller region get released after this change, but I guess the difference is negligible and within noise.

Reply:
> but I guess the difference is negligible and within noise
 
We should measure it rather than guessing. Cheng did measurements when the cold run changes were added to validate the results.

Reply:
Run liberty startup 10 times with cold cache. The RSS (Mean (±Std Dev)) of the metadata is: 
Without this change: 2401 (±335) kbytes. 
With this change: 2298 (±292) kbytes.

Reply:
If I understand the numbers correctly, this is a 100K regression or about 4%. I wonder if we can avoid this regression.

Reply:
The footprint is even lower with this change. But considering the standard deviation is about 300k, I think the 100k is just some noise.

Reply:
ok, I had it backwards.